packages/@groupby/elements-gb-tracker-plugin/src/gb-tracker-plugin.ts

```diff
@@ -0,0 +1,98 @@
+import { Plugin, PluginMetadata, PluginRegistry } from '@groupby/elements-core';
+import { BEACON_SEARCH, BeaconSearchPayload } from '@groupby/elements-events';
+// eslint-disable-next-line import/no-unresolved
+import { AutoSearchEvent } from 'gb-tracker-client/models';
+// eslint-disable-next-line import/no-unresolved
+import { GbTracker } from 'gb-tracker-client/slim-es';
+
+/**
+ * This plugin is responsible for exposing a [[GbTracker]] instance to Core.
+ * It listens for generic beacon events and triggers GroupBy-specific beacons.
+ */
+export default class GbTrackerPlugin implements Plugin {
+  get metadata(): PluginMetadata {
+    return {
+      name: 'gb_tracker',
+      depends: ['dom_events'],
+    };
+  }
+
+  /**
+   * The value that the [[GbTrackerPlugin]] exposes to the Core entity.
+   */
+  gbTracker: ReturnType<typeof GbTracker>;
+
+  /**
+   * A reference to the registry of plugins for later use.
+   */
+  core: PluginRegistry;
+
+  /**
+   * Name of the events plugin.
+   */
+  eventsPluginName = 'dom_events';
+
+  /**
+   * The [[GbTrackerPlugin]] constructor instantiates an instance of the
+   * [[GbTracker]] client and attaches it to this plugin's `gbTracker`
+   * property.
+   *
+   * @param options The options for instantiating the GbTracker client.
```

Comment:
Typedoc assigns a destructured parameter a fixed name, which is `_namedParameters`.

```suggestion
   * @param _namedParameters The options for instantiating the [[GbTracker]] client.
```